### PR TITLE
Reject rounds=0 for SHA1 hashes

### PR DESCRIPTION
**Pull request by rsgowman**
Port of https://github.com/firebase/firebase-admin-node/pull/677

### CODE REVIEW

File: src/test/java/com/google/firebase/auth/UserImportHashTest.java [hunk 1]
```diff
@@ -145,6 +146,91 @@ public void testBasicHash() {
     }
   }
 
+  private void assertBuilderThrowsIllegalArgumentException(Md5.Builder builder) {
```
Comment (rsgowman) [inline src/test/java/com/google/firebase/auth/UserImportHashTest.java]: An alternative to these helpers is to make all of the builders extend from a common interface (which they do: RepeatableHash.Builder) and make it public (which it isn't; and neither is RepeatableHash) even if it's just `@VisibleForTesting`. I didn't do that, in the (naively?) optimistic hope that we'll update to java8 "soon" which will make all that unnecessary.

Comment (hiranya911) [reply src/test/java/com/google/firebase/auth/UserImportHashTest.java]: May be move this to `InvalidHashTest`. There `RepeatableHash` is already visible.

File: src/test/java/com/google/firebase/auth/UserImportHashTest.java [hunk 2]
```diff
@@ -145,6 +146,91 @@ public void testBasicHash() {
...
+    try {
+      builder.build();
+      fail("Expected IllegalArgumentException to be thrown but no expecption occurred.");
+    } catch (IllegalArgumentException expected) {
+    }
+  }
+
+  private void assertBuilderThrowsIllegalArgumentException(Sha1.Builder builder) {
```
Comment (hiranya911) [inline src/test/java/com/google/firebase/auth/UserImportHashTest.java]: If you decide to keep these helpers, move them further down in the file, after all the public methods.

File: src/test/java/com/google/firebase/auth/UserImportHashTest.java (hunk 1 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/UserImportHashTest.java]: Oh, yeah; that's definitely better. (Despite the name, I also used this file to test for valid hashes too.)

File: src/test/java/com/google/firebase/auth/UserImportHashTest.java (hunk 2 above)
Comment (rsgowman) [reply src/test/java/com/google/firebase/auth/UserImportHashTest.java]: (no longer applicable.)

**Pull request merged**